whois-api/src/main/java/net/ripe/db/whois/api/rest/WhoisSearchService.java

```diff
@@ -101,15 +102,18 @@
     private final AccessControlListManager accessControlListManager;
     private final RpslObjectStreamer rpslObjectStreamer;
     private final SourceContext sourceContext;
+    private final String nonAuthSource;
 
     @Autowired
     public WhoisSearchService(
             final AccessControlListManager accessControlListManager,
             final RpslObjectStreamer rpslObjectStreamer,
-            final SourceContext sourceContext) {
+            final SourceContext sourceContext,
+            @Value("${whois.nonauth.source}") final String nonAuthSource) {
```

Comment:
Can you inject whois.nonauth.source into DefaultSourceContext and put logic there?

Reply:
Done.